src/components/student/StudentPage.tsx

```diff
@@ -0,0 +1,53 @@
+'use client'
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
```

Comment:
dont use react-router-dom because next has a built-in navigation library for this. may conflict with it too. check `useSearchParams`.